manifests/config.pp

```diff
@@ -3,89 +3,25 @@
 # This class is called from consul_template for service config.
 #
 class consul_template::config (
-  $consul_host,
-  $consul_port,
-  $consul_token,
-  $consul_retry,
-  $purge = true,
+  $config_hash = {},
+  $purge       = true,
 ) {
-
-  concat::fragment { 'header':
-    target  => 'consul-template/config.json',
-    content => inline_template("consul = \"<%= @consul_host %>:<%= @consul_port %>\"\ntoken = \"<%= @consul_token %>\"\nretry = \"<%= @consul_retry %>\"\n\n"),
-    order   => '00',
-  }
-
-  # Set the log level
-  concat::fragment { 'log_level':
-    target  => 'consul-template/config.json',
-    content => inline_template("log_level = \"${::consul_template::log_level}\"\n"),
-    order   => '01'
-  }
-
-  # Set wait param if specified
-  if $::consul_template::consul_wait {
-    concat::fragment { 'consul_wait':
-      target  => 'consul-template/config.json',
-      content => inline_template("wait = \"${::consul_template::consul_wait}\"\n\n"),
-      order   => '02',
-    }
+  # Using our parent module's pretty_config & pretty_config_indent just because
+  $content = consul_sorted_json($config_hash, $consul::pretty_config, $consul::pretty_config_indent)[0,-1]
```

Comment:
There's a couple places where I'm messing with the json'd with substrings so that I can join together the base config and 1 or more template configs. It's the part of this I'm the least happy about so suggestions welcome, but what's here does work.